### PR TITLE
feat: add datalist for input

### PR DESCRIPTION
**Pull request by nmerget**

### CODE REVIEW

File: packages/components/src/components/input/input.lite.tsx [hunk 1]
```diff
@@ -93,6 +98,9 @@ export default function DBInput(props: DBInputProps) {
 
 	onMount(() => {
 		state._id = props.id ? props.id : 'input-' + uuid();
+		state._dataListId = props.dataListId
+			? props.dataListId
+			: `datalist-${state._id}`;
```
Comment (mfranzke) [inline packages/components/src/components/input/input.lite.tsx:103]:
```suggestion
		state._dataListId = props.dataListId || `datalist-${state._id}`;
```

From my understanding this would be the equivalent.

File: packages/components/src/components/input/input.lite.tsx [hunk 2]
```diff
@@ -122,13 +130,15 @@ export default function DBInput(props: DBInputProps) {
 				aria-labelledby={state._id + '-label'}
 				disabled={props.disabled}
 				required={props.required}
+				defaultValue={props.defaultValue}
```
Comment (mfranzke) [inline packages/components/src/components/input/input.lite.tsx:133]: I'm not aware that an attribute named `defaultValue` exists. What's the expected result (especially, as we already do have `value`)?

File: packages/components/src/components/input/input.lite.tsx [hunk 3]
```diff
@@ -148,6 +158,24 @@ export default function DBInput(props: DBInputProps) {
 			<Show when={state.iconVisible(props.iconAfter)}>
 				<DBIcon icon={props.iconAfter} class="icon-after" />
 			</Show>
+			<DBIcon icon="expand-more" class="datalist-icon" />
```
Comment (mfranzke) [inline packages/components/src/components/input/input.lite.tsx:161]:
```suggestion
```
Let's just set this icon via (S)CSS.

File: packages/components/src/components/input/input.lite.tsx (hunk 2 above)
Comment (nmerget) [reply packages/components/src/components/input/input.lite.tsx:131]: This is a react property. The other framework should ignore it

File: packages/components/src/components/input/input.lite.tsx (hunk 3 above)
Comment (nmerget) [reply packages/components/src/components/input/input.lite.tsx:159]: I guess this is the same topic, we had to discuss. How do we want to handle icons in components?

Comment (mfranzke) [reply packages/components/src/components/input/input.lite.tsx:159]: In general as a preparation to our alignment:
- Icons that we'd like to set based on the guidelines: Via SCSS mixin.
- Icons that the user could set: Either via SCSS mixin, or HTML attribute `data-icon`, depending on their tech stack.

The `db-icon` element was only a workaround for Web Components.

Comment (mfranzke) [reply packages/components/src/components/input/input.lite.tsx]: I've added these aspects and decisions to https://github.com/db-ui/mono/blob/ee3967f31ac60b66b5e349dca34757c4042f86ed/docs/adr-04-icons.md

File: packages/components/src/components/input/input.scss [hunk 4]
```diff
@@ -159,6 +186,10 @@
 		padding: $db-spacing-fixed-xs var(--db-input-padding-inline-end)
 			$db-spacing-fixed-xs var(--db-input-padding-inline-start);
 
+		&::-webkit-calendar-picker-indicator {
+			display: none !important;
+		}
```
Comment (annsch) [inline packages/components/src/components/input/input.scss:191]: @nmerget why did you remove this? Should we use the calendar icon until we added the correct icon?

Comment (nmerget) [reply packages/components/src/components/input/input.scss]: fixed

**Review by annsch — APPROVED** [review]
